app/src/state.js

```diff
@@ -0,0 +1,41 @@
+import React, { useRef, useEffect } from "react";
+import mapboxgl from "!mapbox-gl"; // eslint-disable-line import/no-webpack-loader-syntax
+import Header from "./Header";
+
+export default function State() {
+  const mapContainer = useRef(null);
+  const map = useRef(null);
+  useEffect(() => {
+    if (map.current) return; // initialize map only once
```

Comment:
@ashutoshgt as i understand whenever we are clicking state to poi or poi to state, components are getting rendered completely, which will have `map.current` set to null. What is the need of this condition in current scenario